core/src/com/unciv/logic/map/mapgenerator/RiverGenerator.kt

```diff
@@ -189,4 +190,105 @@ class RiverGenerator(
+                        yield(rightSharedNeighbor?.isConnectedByRiver(tile))
+                        yield(rightSharedNeighbor?.isConnectedByRiver(otherTile))
+                    }.count { it == true }
+                    verticesFormYCount = sequence {
+                        yield(leftSharedNeighbor?.run { isConnectedByRiver(tile) && isConnectedByRiver(otherTile) })
+                        yield(rightSharedNeighbor?.run { isConnectedByRiver(tile) && isConnectedByRiver(otherTile) })
+                    }.count { it == true }
+                }
+            }
+
+            // Collect data
+            val viableNeighbors = mutableListOf<NeighborData>() // includes those we already have a river edge with - need the stats
+            for (clockPosition in 2..12 step 2) {
+                val otherTile = tileMap.getClockPositionNeighborTile(tile, clockPosition)
+                    ?: continue // No tile in that direction
+                if (!otherTile.isLand) continue // No rivers bordering water
+                viableNeighbors += NeighborData(clockPosition, otherTile)
+            }
+            if (viableNeighbors.all { it.isConnectedByRiver }) return false // cannot place another river
+
+            // Greatly encourage connecting to sea unless the tile already has a river to sea, in which case slightly discourage another one
+            val edgeToSeaPriority = if (viableNeighbors.none { it.isConnectedByRiver && it.edgeLeadsToSea }) 9 else -1
+
+            // Assign priorities
+            var maxPriority = Int.MIN_VALUE
+            for (neighbor in viableNeighbors) {
+                neighbor.priority = when {
+                    neighbor.isConnectedByRiver -> -9 // ensures this isn't chosen
+                    neighbor.edgeLeadsToSea -> edgeToSeaPriority + neighbor.connectedRiverCount - 3 * neighbor.verticesFormYCount
+                    // Just 6 possible cases left:
+                    //  * Connect two bends = -2
+                    //  * Connect a bend to nothing = -1
+                    //  * Connect nothing = 0
+                    //  * Connect a bend with an open end = 1
+                    //  * Connect to one open end = 2
+                    //  * Connect two open ends = 3 (make that 4 to simplify)
+                    neighbor.verticesFormYCount == 2 -> -2
+                    neighbor.verticesFormYCount == 1 -> neighbor.connectedRiverCount * 2 - 5
+                    else -> neighbor.connectedRiverCount * 2
+                }
+                if (neighbor.priority > maxPriority) maxPriority = neighbor.priority
+            }
+
+            // Finally - choose
+            val choice = viableNeighbors.filter { it.priority == maxPriority }.random()
+            tile.setConnectedByRiver(choice.otherTile, choice.clockPosition)
+            return true
+        }
+
+        private fun Tile.setConnectedByRiver(otherTile: Tile, clockPosition: Int) {
```

Comment:
If we know the neighbor, we can get the clock position from that - having two inputs implies cases where the position is not the relative neighbor position?